source/simulator/stokes_matrix_free.cc

```diff
@@ -1606,6 +1674,126 @@ namespace aspect
+                        = derivatives->viscosity_derivative_wrt_strain_rate[q];
+
+                      // use the spd factor when the stabilization is PD or SPD
+                      const double alpha =  (sim.newton_handler->parameters.velocity_block_stabilization
+                                             & Newton::Parameters::Stabilization::PD)
+                                            != Newton::Parameters::Stabilization::none
+                                            ?
+                                            Utilities::compute_spd_factor<dim>(out.viscosities[q], strain_rate, viscosity_derivative_wrt_strain_rate,
+                                                                               sim.newton_handler->parameters.SPD_safety_factor)
+                                            :
+                                            1.0;
+
+                      active_viscosity_derivative_wrt_pressure_table(cell,q)[i]
+                        = viscosity_derivative_wrt_pressure * newton_derivative_scaling_factor;
+
+                      for (unsigned int m=0; m<dim; ++m)
+                        for (unsigned int n=0; n<dim; ++n)
+                          {
+                            active_strain_rate_table(cell, q)[m][n][i]
+                              = strain_rate[m][n];
+
+                            active_viscosity_derivative_wrt_strain_rate_table(cell, q)[m][n][i]
+                              = viscosity_derivative_wrt_strain_rate[m][n]
+                                * newton_derivative_scaling_factor * alpha;
+                          }
+                    }
+                }
+            }
+
+          // symmetrize when the stabilization is symmetric or SPD
+          const bool is_symmetric =
+            (sim.newton_handler->parameters.velocity_block_stabilization & Newton::Parameters::Stabilization::symmetric)
+            != Newton::Parameters::Stabilization::none;
+
+          stokes_matrix.fill_Newton_cell_data(active_viscosity_derivative_wrt_pressure_table,
+                                              active_strain_rate_table,
+                                              active_viscosity_derivative_wrt_strain_rate_table,
+                                              is_symmetric);
+        }
+      else
+        {
+          active_strain_rate_table.reinit(TableIndices<2>(0,0));
+          active_viscosity_derivative_wrt_pressure_table.reinit(TableIndices<2>(0,0));
+          active_viscosity_derivative_wrt_strain_rate_table.reinit(TableIndices<2>(0,0));
+
+          stokes_matrix.fill_Newton_cell_data(active_viscosity_derivative_wrt_pressure_table,
+                                              active_strain_rate_table,
+                                              active_viscosity_derivative_wrt_strain_rate_table,
+                                              false /*Doesn't matter, it's not going to be used*/);
```

Comment:
Remove this comment here and put it at the beginning of this `else` branch as something like the following: `// Hand over empty tables, because they will not be used`